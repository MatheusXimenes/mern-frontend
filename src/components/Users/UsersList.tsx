import React, { FC } from "react";

import { IUser } from "../../models/UserModel";
import UserItem from "./UserItem";
import Card from "../UIElements/Card";

import "./UsersList.css";

type UserListProps = {
  users: IUser[];
};

const UsersList: FC<UserListProps> = ({ users }) => {
  if (users.length === 0) {
    return (
      <div className="center">
        <Card>
          <h2>No users found.</h2>
        </Card>
      </div>
    );
  }

  return (
    <ul className="users-list">
      {users.map((user) => (
        <UserItem
          key={user._id}
          id={user._id}
          image={user.imageUrl}
          name={user.name}
          placeCount={user.places.length}
        />
      ))}
    </ul>
  );
};

export default UsersList;
