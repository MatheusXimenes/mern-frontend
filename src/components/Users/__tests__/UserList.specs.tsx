import React, {
  render,
  screen,
  fireEvent,
  getByRole,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { BrowserRouter as BRouter, Router, Route } from "react-router-dom";
import UsersList from "../UsersList";

const MockData = [
  {
    _id: 1,
    name: "Matheus",
    email: "mattximca@gmail.com",
    password: "dfjhgaslkfjhs",
    imageUrl: "Url",
    places: [],
  },
];

describe("UserList Component", () => {
  describe("Rendering", () => {
    it("No Users Found Warning", () => {
      render(<UsersList users={[]} />);
      screen.getByRole("heading", { level: 2, name: "No users found." });
    });

    it("List of UserItems", () => {
      render(
        <BRouter>
          <Route>
            <UsersList users={MockData} />
          </Route>
        </BRouter>
      );
      const linkEl = screen.getAllByRole("link");
      expect(linkEl.length).toBe(1);
    });
  });

  describe("ScreenShot", () => {
    it("No Users Found Warning", () => {
      const tree = renderer.create(<UsersList users={[]} />);
      expect(tree).toMatchSnapshot();
    });

    it("List of UserItems", () => {
      const tree = renderer.create(
        <BRouter>
          <Route>
            <UsersList users={MockData} />
          </Route>
        </BRouter>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
