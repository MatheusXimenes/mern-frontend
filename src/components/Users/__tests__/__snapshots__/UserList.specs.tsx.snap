// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList Component ScreenShot List of UserItems 1`] = `
<ul
  className="users-list"
>
  <li
    className="user-item"
    data-testid={1}
  >
    <div
      className="card user-item__content"
    >
      <a
        href="/1/places"
        onClick={[Function]}
      >
        <div
          className="user-item__image"
        >
          <div
            className="avatar undefined"
          >
            <img
              alt="Matheus"
              src="Url"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </div>
        </div>
        <div
          className="user-item__info"
        >
          <h2>
            Matheus
          </h2>
          <h3>
            0
             
            Places
          </h3>
        </div>
      </a>
    </div>
  </li>
</ul>
`;

exports[`UserList Component ScreenShot No Users Found Warning 1`] = `
<div
  className="center"
>
  <div
    className="card undefined"
  >
    <h2>
      No users found.
    </h2>
  </div>
</div>
`;
